service-mvt-wfs/src/main/java/org/oskari/service/mvt/wfs/OskariCachingSchemaLocator.java

```diff
@@ -0,0 +1,57 @@
+package org.oskari.service.mvt.wfs;
+
+
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.util.IOHelper;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl;
+import org.eclipse.xsd.XSDSchema;
+import org.eclipse.xsd.util.XSDResourceImpl;
+import org.eclipse.xsd.util.XSDSchemaLocator;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class OskariCachingSchemaLocator implements XSDSchemaLocator {
+
+    private static final Logger LOG = LogFactory.getLogger(OskariCachingSchemaLocator.class);
+    private static final Map<String, XSDSchema> CACHE = new ConcurrentHashMap<String, XSDSchema>();
```

Comment:
Should this use ComputeOnceCache as well with a big limit and long expiration?

Reply:
That way we could programmatically flush the cache.

Reply:
Yes